src/fsearch_window_actions.c

```diff
@@ -584,6 +586,30 @@ fsearch_window_action_open_folder(GSimpleAction *action, GVariant *variant, gpoi
     fsearch_window_action_open_generic(self, true, false);
 }
 
+static void
+fsearch_window_action_preview(GSimpleAction *action, GVariant *variant, gpointer user_data) {
+    FsearchApplicationWindow *self = user_data;
+    guint xid = 0;
+    GList *file_list = NULL;
+
+    const guint selected_rows = fsearch_application_window_get_num_selected(self);
+    if (selected_rows == 0) {
+        return;
+    }
+
+    fsearch_application_window_selection_for_each(self, prepend_full_path_to_list, &file_list);
+    file_list = g_list_reverse(file_list);
+
+    GtkWidget *toplevel = gtk_widget_get_toplevel(GTK_WIDGET(self));
+#ifdef GDK_WINDOWING_X11
```

Comment:
I suppose the `xid` is used to tell GTK how to position the preview window properly above the FSearch window? Do you know if it's possible to do the same for Wayland sessions?

Reply:
Yes, but I don't know how to do that for Wayland sessions :(

Reply:
No problem. I just had a look at how Nautilus handles this and they don't have a workaround for Wayland either, so it's likely not so trivial or maybe not even possible at all.